src/irr/asset/CImageWriterPNG.cpp

```diff
@@ -103,30 +149,45 @@ bool CImageWriterPNG::writeAsset(io::IWriteFile* _file, const SAssetWriteParams&
 		png_destroy_write_struct(&png_ptr, &info_ptr);
 		return false;
 	}
+
+	core::smart_refctd_ptr<ICPUImage> convertedImage;
+	{
+		const auto channelCount = asset::getFormatChannelCount(image->getCreationParameters().format);
+		if (channelCount == 1)
+			convertedImage = getPNGConvertedOutput<asset::EF_R8_SRGB>(image);
+		else if(channelCount == 2 || channelCount == 3)
+			convertedImage = getPNGConvertedOutput<asset::EF_R8G8B8_SRGB>(image);
+		else
+			convertedImage = getPNGConvertedOutput<asset::EF_R8G8B8A8_SRGB>(image);
+	}
 	
-	auto format = image->getColorFormat();
+	const auto& convertedImageParams = convertedImage->getCreationParameters();
+	const auto& convertedRegion = convertedImage->getRegions().begin();
+	auto convertedFormat = convertedImageParams.format;
+
+	asset::TexelBlockInfo blockInfo(convertedFormat);
+	core::vector3du32_SIMD trueExtent = blockInfo.convertTexelsToBlocks(convertedRegion->getTexelStrides());
```

Comment:
dont call `convertTexelToBlocks` its confusing as fuck and its a NOOP for a PNG image

Reply:
or make the blockInfo at all

Reply:
blockInfo is better to me since final true extent take pitch into account since your functions deals with row buffer length afaik

Reply:
to fix

Reply:
no point of this being here at all, you can get it all from the first region